dbflow/src/main/java/com/raizlabs/android/dbflow/structure/ModelAdapter.java

```diff
@@ -63,12 +57,8 @@ public DatabaseStatement getInsertStatement(DatabaseWrapper databaseWrapper) {
      * @return The precompiled full statement for this table model adapter
      */
     public DatabaseStatement getCompiledStatement() {
-        if (compiledStatement == null) {
-            compiledStatement = getCompiledStatement(
-                    FlowManager.getDatabaseForTable(getModelClass()).getWritableDatabase());
-        }
-
-        return compiledStatement;
+        return getCompiledStatement(
```

Comment:
Well, maybe I got your comment wrong, but not caching the compile statement was the whole point of this PR. ;) Otherwise `finalize()` is called which then closes the statement later on as described in #706. But probably I just misunderstood your remark.